cocotb/triggers.py

```diff
@@ -673,13 +654,13 @@ def __repr__(self):
         )
 
 
-@decorators.coroutine
-def _wait_callback(trigger, callback):
+async def _wait_callback(trigger, callback):
     """
-    Wait for a trigger, and call `callback` with the outcome of the yield.
+    Wait for a trigger, and call `callback` with the outcome of the await.
     """
+    trigger = cocotb.scheduler._trigger_from_any(trigger)
     try:
-        ret = outcomes.Value((yield trigger))
+        ret = outcomes.Value(await trigger)
```

Comment:
Regarding my earlier remark about the behavior change here on `Waitable` objects - I think since we only really have three `Waitable` types in the wild, and none of them leave much behind to clean up, this is probably ok.